### DISCUSSION TITLE
How does reset work

### DISCUSSION THREAD

**Discussion by jasmine-anzx** [jesseduffield/lazygit#3991] [Q&A]
I select a commit, press 'g', popup showed asking if I want soft, hard or mixed reset. I pressed 's' to choose soft reset. Nothing happened.

I tried again by moving cursor to select soft commit then Enter, nothing happen as well. What did I do wrong? Thank you

**Comment by mark2185** [comment]
That depends which commit did you select, but it works the same as `git reset --soft`, it should move the `HEAD` to the selected commit, and put all of the changes between the old and the new `HEAD` into the `staging` state.

Could you please give some steps to reproduce what you're experiencing? Or a recording?

**Comment by emmet-clarke** [comment]
Yes this doesn't work for me either. Let's say I accidentally commit to local master. I navigate to the commits pane, highlight the commit in question, press g and I get a window asking me if I want a mixed, soft, or hard reset. With the indicated kesy of m, s, or h. None of these keys does anything nor does highlighting one of them and pressing enter.

**Reply by mark2185** [reply]
Is this on the latest version? Are you resetting over merge commits by any chance?

**Comment by felixoakz** [comment]
Same here. Neither of the resets are working via Lazy GUI. Nothing happens when choosing r + any of the options.
But if i go to the terminal and run ```$git reset --mixed HEAD-1``` it works normally.

**Reply by stefanhaller** [reply]
This seems to be a common misconception. When resetting, you need to select the commit that you want to reset to, not the one that you want to get rid of.

**Reply by felixoakz** [reply]
You're absolutely right — my bad, guess I wasn’t thinking straight. Thanks for the clarification!

**Comment by WinnerWind** [comment]
You need to select a commit from the commits menu before pressing `g`. When resetting, you need to select the commit that you want to reset TO. In short, resetting will send your working directory tree back to what it was during *that commit*'s state.

Of course, as with all git operations, be mindful of what it is doing, and be sure to stash your changes if you're going to do a hard reset.

EDIT: Resetting from the Files menu will attempt to reset your tree to the **state of the upstream**. Resetting from the Branches menu will attempt to reset your tree to the **state of the selected branch**. Resetting from the Commits menu will attempt to reset your tree to the **state of the selected commit**.

**Reply by RizaHKhan** [reply]
This. 

Say I want to reset this commit:
<img width="390" height="106" alt="image" src="https://github.com/user-attachments/assets/fbb77aa2-14bd-436a-b652-ed24f49ffb4e" />

I click `g` on this commit:

<img width="452" height="102" alt="image" src="https://github.com/user-attachments/assets/91ee1579-5eab-474e-b198-367a96d7f77a" />
